test/io/test_octree_compression.cpp

```diff
@@ -48,102 +48,182 @@ int total_runs = 0;
 char* pcd_file;
 
 #define MAX_POINTS 10000.0
-#define MAX_XYZ 1024.0
 #define MAX_COLOR 255
-#define NUMBER_OF_TEST_RUNS 2
+#define NUMBER_OF_TEST_RUNS 3
 
 TEST (PCL, OctreeDeCompressionRandomPointXYZRGBA)
 {
   srand(static_cast<unsigned int> (time(NULL)));
+  for (const double MAX_XYZ : {1.0, 1024.0}) { // Small clouds, large clouds
+    // iterate over all pre-defined compression profiles
+    for (int compression_profile = pcl::io::LOW_RES_ONLINE_COMPRESSION_WITHOUT_COLOR;
+      compression_profile != pcl::io::COMPRESSION_PROFILE_COUNT; ++compression_profile) {
+      // instantiate point cloud compression encoder/decoder
+      pcl::io::OctreePointCloudCompression<pcl::PointXYZRGBA> pointcloud_encoder((pcl::io::compression_Profiles_e) compression_profile, false);
+      pcl::io::OctreePointCloudCompression<pcl::PointXYZRGBA> pointcloud_decoder;
+      pcl::PointCloud<pcl::PointXYZRGBA>::Ptr cloud_out(new pcl::PointCloud<pcl::PointXYZRGBA>());
+      // iterate over runs
+      for (int test_idx = 0; test_idx < NUMBER_OF_TEST_RUNS; test_idx++, total_runs++)
+      {
+        try
+        {
+          int point_count;
+          do { // empty point cloud hangs decoder
+            point_count = MAX_POINTS * rand() / RAND_MAX;
+          } while (point_count < 1);
+          // create shared pointcloud instances
+          pcl::PointCloud<pcl::PointXYZRGBA>::Ptr cloud(new pcl::PointCloud<pcl::PointXYZRGBA>());
+          // assign input point clouds to octree
+          // create random point cloud
+          for (int point = 0; point < point_count; point++)
+          {
+            // generate a random point
+            pcl::PointXYZRGBA new_point;
+            new_point.x = static_cast<float> (MAX_XYZ * rand() / RAND_MAX);
+            new_point.y = static_cast<float> (MAX_XYZ * rand() / RAND_MAX),
+            new_point.z = static_cast<float> (MAX_XYZ * rand() / RAND_MAX);
+            new_point.r = static_cast<int> (MAX_COLOR * rand() / RAND_MAX);
+            new_point.g = static_cast<int> (MAX_COLOR * rand() / RAND_MAX);
+            new_point.b = static_cast<int> (MAX_COLOR * rand() / RAND_MAX);
+            new_point.a = static_cast<int> (MAX_COLOR * rand() / RAND_MAX);
```

Comment:
Use point cloud generator?

Reply:
`CloudGenerator` only fills out x, y, z, but no rgb, or did I miss something? Then I would additionally have to use a `UniformGenerator` for r, g, b.

Reply:
🤦🏾 Just saw that